src/tests/unittests/configuration_helper/adapters/test_uhfli_instrument_adapter.py

```diff
@@ -0,0 +1,44 @@
+import unittest
+from unittest.mock import patch, MagicMock
+
+import numpy as np
+import zhinst
+
+from qilib.configuration_helper import InstrumentAdapterFactory
+
+
+class TestZIUHFLIInstrumentAdapter(unittest.TestCase):
+    def test_read_apply(self):
+        with patch.object(zhinst.utils, 'create_api_session', return_value=3 * (MagicMock(),)):
+            adapter = InstrumentAdapterFactory.get_instrument_adapter('ZIUHFLIInstrumentAdapter', 'dev4242')
+        adapter.instrument.scope_trig_level(0.2)
+        trigger_level = adapter.instrument.scope_trig_level.raw_value
+        self.assertEqual(0.2, trigger_level)
+        config = adapter.read()
+
+        adapter.instrument.scope_trig_level(0.7)
+        trigger_level = adapter.instrument.scope_trig_level.raw_value
+        self.assertEqual(0.7, trigger_level)
+
+        adapter.apply(config)
+        trigger_level = adapter.instrument.scope_trig_level.raw_value
+        self.assertEqual(0.2, trigger_level)
```

Comment:
```
adapter.instrument.close()
```

The unittest give a warning when not closing a QCoDeS instrument at the end of this test.

Reply:
✔️